src/loader.hpp

```diff
@@ -0,0 +1,22 @@
+#ifndef _LOADER_HPP_
+#define _LOADER_HPP_
+
+#include "datatype.hpp"
+#include <fstream>
+#include <iostream>
+
+class InstructionLoader {
+  public:
+    InstructionLoader(const std::string &filename);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Class 'InstructionLoader' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/app/jerry0715no14/Cache_sim/pullRequest?prid=4720103)